cpp/ql/test/experimental/query-tests/Security/CWE/CWE-758/semmle/tests/test.cpp

```diff
@@ -0,0 +1,218 @@
+#define NULL (0)
+typedef unsigned long size_t;
+typedef struct {} FILE;
+FILE *ef;
+void *malloc(size_t size);
+size_t strlen(const char *str);
+void free(void *ptr);
+unsigned long copy_from_user (void * to, void * from, unsigned long n);
+int fread(char *buf, int size, int count, FILE *fp);
+int getSize3();
+int getLen3(char *s);
+
+int getSize2()
+{
+  int m;
+  m = getSize3();
+  if (m<0)  return 0;
+  return m;
+}
+
+int getSize1()
+{
+  int m;
+  m = getSize3();
+  if (m<0) m=0;
+  return m;
+}
+int getLen4() {
+  char c[1];
+  fread(c, 1, 1, ef);
+  return (int)*c;
+}
+int getLen2()
+{
+  char str[4] = {0xff, 0xff, 0xff, 0xff};
+  fread(str, 1, 4, ef);
+  return getLen3(str);
+}
+void getLen1(int * len)
+{
+  *len = getLen2();
+}
+
+
+void goodTest1()
+{
+  char *ptr;
+  int len = 10;
+  ptr = (char *)malloc(len); // GOOD
+  if(ptr) {
+	 ptr[len-1] = 0;
+	 free(ptr);
+  }
```

Comment:
This is leaking the allocation from line 49. Is that intended?

This happens all throughout this test file.

Reply:
differs only in this.
this test case is required, it covers detection when used in a function, size

Reply:
I don't see leak.
could you show it in more detail.